standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -3799,6 +3798,29 @@ public Partition exchange_partition(Map<String, String> partitionSpecs,
       return new Partition();
     }
 
+    private boolean isRenameAllowed(String catalogName, String srcDBName, String destDBName)
+            throws MetaException, NoSuchObjectException {
+      RawStore ms = getMS();
+      if (!srcDBName.equalsIgnoreCase(destDBName)) {
+        Database destDB = ms.getDatabase(catalogName, destDBName);
+        if (destDB == null) {
```

Comment:
ms.getDatabase will never return null instead throw NoSuchObjectException. No need to handle it anywhere.